### PR TITLE
Add telemetry bi to azure spring cloud website, #5.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Add event for website access.

* Add event for project generate.

* Add event for account login.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java [hunk 1]
```diff
@@ -75,22 +101,52 @@ private void addBuildInformation(@NonNull Map<String, Object> model) {
...
+    public String home(Map<String, Object> model, HttpServletRequest request) {
+
         this.addBuildInformation(model);
         this.renderHome(model);
+        this.triggerAccessEvent(request.getRemoteAddr());
 
         return "home";
     }
 
     @RequestMapping("/starter.zip")
```
Comment (yaweiw) [inline src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java]: change /starter.zip to something else e.g. /microservice.zip?

File: src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java [hunk 2]
```diff
@@ -120,12 +176,14 @@ public String home(Map<String, Object> model) {
 
     @RequestMapping(path = "/starter.tgz", produces = "application/x-compress")
```
Comment (yaweiw) [inline src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java]: same

File: src/main/resources/ApplicationInsights.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings" schemaVersion="2014-05-30">
+    <!-- The key from the portal: -->
+    <InstrumentationKey>d3278c59-22d7-4697-af8c-ff0077e3b4e9</InstrumentationKey>
```
Comment (yaweiw) [inline src/main/resources/ApplicationInsights.xml]: By exposing the key, will it render security risk?

File: src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java]: copy that.

File: src/main/resources/ApplicationInsights.xml (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/main/resources/ApplicationInsights.xml]: No security risk, this only used to send data. But if others may abuse this key. From spring-data-cosmosdb, it use placeholder here, and patch the real key when release (that means you can find the key from release package). But here we are web application different to release package. So I just put the key here.

File: src/main/java/com/microsoft/azure/springcloudplayground/util/TelemetryProxy.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+public class TelemetryProxy {
+
+    private static final String PROJECT_PROPERTY_FILE = "/META-INF/project.properties";
+
+    private final TelemetryClient client;
+    private final PropertyLoader loader;
+
+    public TelemetryProxy() {
+        this.client = new TelemetryClient();
+        this.loader = new PropertyLoader(PROJECT_PROPERTY_FILE);
```
Comment (yaweiw) [inline src/main/java/com/microsoft/azure/springcloudplayground/util/TelemetryProxy.java]: loader is just used once. if it's big better pass in the value when constructing the class

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/springcloudplayground/util/TelemetryProxy.java]: Yes, we can make it static here.

**Pull request merged**